sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala

```diff
@@ -16,84 +16,42 @@
 
 package com.nvidia.spark.rapids
 
-import ai.rapids.cudf.Scalar
 import com.nvidia.spark.rapids.RapidsPluginImplicits._
 
 import org.apache.spark.sql.catalyst.analysis.{TypeCheckResult, TypeCoercion}
 import org.apache.spark.sql.catalyst.expressions.{ComplexTypeMergingExpression, Expression}
 import org.apache.spark.sql.types.{BooleanType, DataType}
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
-abstract class GpuConditionalExpression extends ComplexTypeMergingExpression with GpuExpression {
+trait GpuConditionalExpression extends ComplexTypeMergingExpression with GpuExpression {
 
   protected def computeIfElse(
       batch: ColumnarBatch,
-      predicateExpr: Expression,
+      predExpr: Expression,
       trueExpr: Expression,
-      falseValues: GpuColumnVector): GpuColumnVector = {
-    withResource(GpuExpressionsUtils.columnarEvalToColumn(predicateExpr, batch)) { predicate =>
-      val trueResult: Any = trueExpr.columnarEval(batch)
-      try {
-        val result = trueResult match {
-          case t: GpuColumnVector => predicate.getBase.ifElse(t.getBase, falseValues.getBase)
-          case t: GpuScalar => predicate.getBase.ifElse(t.getBase, falseValues.getBase)
-          case u =>
-            throw new IllegalStateException(s"Unexpected inputs $u")
-        }
-        GpuColumnVector.from(result, dataType)
-      } finally {
-        trueResult match {
-          case a: AutoCloseable => a.close()
-          case _ =>
-        }
-      }
-    }
-  }
-
-  protected def computeIfElse(
-      batch: ColumnarBatch,
-      predicateExpr: Expression,
-      trueExpr: Expression,
-      falseValue: Scalar): GpuColumnVector = {
-    withResource(GpuExpressionsUtils.columnarEvalToColumn(predicateExpr, batch)) { predicate =>
-      val trueResult: Any = trueExpr.columnarEval(batch)
-      try {
-        val result = trueResult match {
-          case t: GpuColumnVector => predicate.getBase.ifElse(t.getBase, falseValue)
-          case t: GpuScalar => predicate.getBase.ifElse(t.getBase, falseValue)
-          case u =>
-            throw new IllegalStateException(s"Unexpected inputs $u")
-        }
-        GpuColumnVector.from(result, dataType)
-      } finally {
-        trueResult match {
-          case a: AutoCloseable => a.close()
-          case _ =>
+      falseValue: Any): GpuColumnVector = {
+    withResource(GpuExpressionsUtils.columnarEvalToColumn(predExpr, batch)) { pred =>
```

Comment:
This fixed an issue with how we deal with the predicate expression (It cannot be a literal).
Could you please remove the existing checks in GpuOverrides and add in some tests to make sure it is working?

1. https://github.com/NVIDIA/spark-rapids/blob/203769fc3669cd8e0ed76c152a29f01f451e112d/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L1741-L1743
2. https://github.com/NVIDIA/spark-rapids/blob/203769fc3669cd8e0ed76c152a29f01f451e112d/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L1700-L1703

Reply:
Done.

I also found if the `predicate` in `If` or `CaseWhen` is a single literal, seems the expression will be optimized, leaving only an `Alias`. 
Expression tree for  `'IF(a, b, c)'` .
```Bash
  *Expression <Alias> if (a#696) b#697 else c#698 AS (IF(a, b, c))#705 will run on GPU
    *Expression <If> if (a#696) b#697 else c#698 will run on GPU
```

Expression tree for `'IF(TRUE, b, c)'`.
```Bash
*Expression <Alias> b#697 AS (IF(true, b, c))#702 will run on GPU
```